source/include/azure_iot_hub_client_properties.h

```diff
@@ -0,0 +1,310 @@
+                                                                            AzureIoTJSONReader_t * pxJSONReader,
+                                                                            AzureIoTHubMessageType_t xResponseType,
+                                                                            uint32_t * pulVersion );
+
+/**
+ * @brief Property type
+ *
+ */
+typedef enum AzureIoTHubClientPropertyType_t
+{
+    /** @brief Property was originally reported from the device. */
+    eAzureIoTHubClientReportedFromDevice = AZ_IOT_HUB_CLIENT_PROPERTY_REPORTED_FROM_DEVICE,
+    /** @brief Property was received from the service. */
+    eAzureIoTHubClientPropertyWriteable = AZ_IOT_HUB_CLIENT_PROPERTY_WRITEABLE
+} AzureIoTHubClientPropertyType_t;
+
+/**
+ * @brief Iteratively read the Azure IoT Plug and Play component properties.
+ *
+ * Note that between calls, the #az_span pointed to by \p out_component_name shall not be modified,
+ * only checked and compared. Internally, the #az_span is only changed if the component name changes
+ * in the JSON document and is not necessarily set every invocation of the function.
+ *
+ * On success, the `pxJSONReader` will be set on a valid property name. After checking the
+ * property name, the reader can be advanced to the property value by calling
+ * az_json_reader_next_token(). Note that on the subsequent call to this API, it is expected that
+ * the json reader will be placed AFTER the read property name and value. That means that after
+ * reading the property value (including single values or complex objects), the user must call
+ * az_json_reader_next_token().
+ *
+ * Below is a code snippet which you can use as a starting point:
+ *
+ * @code
+ *
+ * while (az_result_succeeded(az_iot_hub_client_properties_get_next_component_property(
+ *       &hub_client, &jr, xResponseType, AZ_IOT_HUB_CLIENT_PROPERTY_WRITEABLE, &pucComponentName)))
+ * {
+ *   // Check if property is of interest (substitute user_property for your own property name)
+ *   if (az_json_token_is_text_equal(&jr.token, user_property))
+ *   {
+ *     az_json_reader_next_token(&jr);
+ *
+ *     // Get the property value here
+ *     // Example: az_json_token_get_int32(&jr.token, &user_int);
+ *
+ *     // Skip to next property value
+ *     az_json_reader_next_token(&jr);
+ *   }
+ *   else
+ *   {
+ *     // The JSON reader must be advanced regardless of whether the property
+ *     // is of interest or not.
+ *     az_json_reader_next_token(&jr);
+ *
+ *     // Skip children in case the property value is an object
+ *     az_json_reader_skip_children(&jr);
+ *     az_json_reader_next_token(&jr);
+ *   }
+ * }
+ *
+ * @endcode
+ *
+ * @warning If you need to retrieve more than one \p xPropertyType, you should first complete the
+ * scan of all components for the first property type (until the API returns
+ * #AZ_ERROR_IOT_END_OF_PROPERTIES). Then you must call az_json_reader_init() again after this call
```

Comment:
`AZ_ERROR_IOT_END_OF_PROPERTIES`: this is currently wrong but will be added with the new results header